lang/java/reef-bridge-proto-java/src/main/java/org/apache/reef/bridge/driver/service/grpc/GRPCDriverServiceConfigurationProvider.java

```diff
@@ -0,0 +1,163 @@
+
+  @Inject
+  private GRPCDriverServiceConfigurationProvider() {
+  }
+
+
+  @Override
+  public Configuration getDriverServiceConfiguration(
+      final ClientProtocol.DriverClientConfiguration driverConfiguration) {
+    Configuration driverServiceConfiguration = DriverServiceConfiguration.CONF
+        .set(DriverServiceConfiguration.DRIVER_SERVICE_IMPL, GRPCDriverService.class)
+        .set(DriverServiceConfiguration.DRIVER_CLIENT_COMMAND, driverConfiguration.getDriverClientLaunchCommand())
+        .build();
+    return driverConfiguration.getDriverRestartEnable() ?
+        Configurations.merge(
+            driverServiceConfiguration,
+            getDriverRestartConfiguration(driverConfiguration),
+            getDriverConfiguration(driverConfiguration),
+            getTcpPortRangeConfiguration(driverConfiguration)) :
+        Configurations.merge(
+            driverServiceConfiguration,
+            getDriverConfiguration(driverConfiguration),
+            getTcpPortRangeConfiguration(driverConfiguration));
+  }
+
+  private Configuration getTcpPortRangeConfiguration(
+      final ClientProtocol.DriverClientConfiguration driverClientConfigurationProto) {
+    JavaConfigurationBuilder configurationModuleBuilder = Tang.Factory.getTang().newConfigurationBuilder()
+        .bindSetEntry(DriverConfigurationProviders.class, TcpPortConfigurationProvider.class);
+    // Setup TCP constraints
+    if (driverClientConfigurationProto.getTcpPortRangeBegin() > 0) {
+      configurationModuleBuilder = configurationModuleBuilder
+          .bindNamedParameter(TcpPortRangeBegin.class,
+              Integer.toString(driverClientConfigurationProto.getTcpPortRangeBegin()));
+    }
+    if (driverClientConfigurationProto.getTcpPortRangeCount() > 0) {
+      configurationModuleBuilder = configurationModuleBuilder
+          .bindNamedParameter(TcpPortRangeCount.class,
+              Integer.toString(driverClientConfigurationProto.getTcpPortRangeCount()));
+    }
+    if (driverClientConfigurationProto.getTcpPortRangeTryCount() > 0) {
+      configurationModuleBuilder = configurationModuleBuilder
+          .bindNamedParameter(TcpPortRangeCount.class,
+              Integer.toString(driverClientConfigurationProto.getTcpPortRangeTryCount()));
+    }
+    return configurationModuleBuilder.build();
+  }
+
+  private Configuration getDriverConfiguration(
+      final ClientProtocol.DriverClientConfiguration driverConfiguration) {
+    ConfigurationModule driverServiceConfigurationModule = DriverServiceConfiguration.STATIC_DRIVER_CONF_MODULE
+        .set(DriverConfiguration.DRIVER_IDENTIFIER, driverConfiguration.getJobid());
+
+    // Set file dependencies
+    final List<String> localLibraries = new ArrayList<>();
+    localLibraries.add(EnvironmentUtils.getClassLocation(GRPCDriverService.class));
+    if (driverConfiguration.getLocalLibrariesCount() > 0) {
```

Comment:
can remove the `if` statement: there is no need to check for `.getLocalLibrariesCount()` - the call `localLibraries.addAll()` will work correctly even when the input list is empty.